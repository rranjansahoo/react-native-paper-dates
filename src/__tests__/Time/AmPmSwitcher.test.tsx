import * as React from 'react'

import renderer from 'react-test-renderer'
import { inputTypes } from '../../Time/timeUtils'
import AmPmSwitcher from '../../Time/AmPmSwitcher'

it('renders AmPmSwitcher', () => {
  const tree = renderer
    .create(
      <AmPmSwitcher
        hours={12}
        onChange={() => null}
        inputType={inputTypes.keyboard}
      />
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
