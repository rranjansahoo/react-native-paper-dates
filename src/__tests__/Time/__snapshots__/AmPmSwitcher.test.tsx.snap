// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AmPmSwitcher 1`] = `
<View
  style={
    [
      {
        "borderWidth": 1,
        "overflow": "hidden",
        "width": 52,
      },
      {
        "borderColor": "rgba(121, 116, 126, 1)",
        "borderRadius": 8,
        "height": 72,
        "marginBottom": 24,
      },
    ]
  }
>
  <View
    accessibilityComponentType="button"
    accessibilityLabel="AM"
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityTraits={
      [
        "button",
        "disabled",
      ]
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        false,
        {
          "flex": 1,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          },
          {
            "backgroundColor": "rgba(234, 221, 255, 1)",
          },
        ]
      }
    >
      <Text
        maxFontSizeMultiplier={1.5}
        selectable={false}
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontWeight": "400",
              "letterSpacing": 0,
            },
            {
              "writingDirection": "ltr",
            },
            [
              {
                "color": "rgba(73, 69, 79, 1)",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": "500",
                "letterSpacing": 0.15,
                "lineHeight": 24,
              },
            ],
          ]
        }
      >
        AM
      </Text>
    </View>
  </View>
  <View
    style={
      [
        {
          "height": 1,
          "width": 52,
        },
        {
          "backgroundColor": "rgba(121, 116, 126, 1)",
        },
      ]
    }
  />
  <View
    accessibilityComponentType="button"
    accessibilityLabel="PM"
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityTraits="button"
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        false,
        {
          "flex": 1,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          },
          {
            "backgroundColor": "rgba(255, 251, 254, 1)",
          },
        ]
      }
    >
      <Text
        maxFontSizeMultiplier={1.5}
        selectable={false}
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontWeight": "400",
              "letterSpacing": 0,
            },
            {
              "writingDirection": "ltr",
            },
            [
              {
                "color": "rgba(73, 69, 79, 1)",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": "500",
                "letterSpacing": 0.15,
                "lineHeight": 24,
              },
            ],
          ]
        }
      >
        PM
      </Text>
    </View>
  </View>
</View>
`;
